ProjFS.Mac/PrjFSKextTests/MockProc.hpp

```diff
@@ -0,0 +1,14 @@
+#include "../PrjFSKext/kernel-header-wrappers/vnode.h"
+
+enum
+{
+    KAUTH_RESULT_ALLOW = 1,
+    KAUTH_RESULT_DENY,
+    KAUTH_RESULT_DEFER
+};
+
+extern "C" int proc_pid(proc_t);
+extern "C" void proc_name(int pid, char * buf, int size);
+extern "C" proc_t vfs_context_proc(vfs_context_t ctx);
+
+void SetProcName(const char* procName);
```

Comment:
Forgive my rusty C++, but if we don that can we call "SetProcName("Finder")"?

Reply:
Yep, if we make the parameter `const std::string&` we can still call the function with `char*`.

This is allowed because `std::string` has a [converting constructor](https://stackoverflow.com/questions/15077466/what-is-a-converting-constructor-in-c-what-is-it-for) for [char*](http://www.cplusplus.com/reference/string/string/string/).  A temporary `std::string` will be automatically constructed for us from the `char*`.

Reply:
Yeah, `const std::string&` is OK here. It's definitely less efficient, but we don't care about that much in the tests, and allows us to pass either `const char*` or `std::string` objects in.